gengokit/generator/gen_test.go

```diff
@@ -346,82 +340,79 @@ func TestUpdateMethods(t *testing.T) {
-		}
+	firstCode, err := renderService(svc, "", te, templPath)
+	if err != nil {
+		t.Fatal(err)
+	}
 
-		svc.Methods = append(svc.Methods, allMethods[1])
+	secondCode, err := renderService(svc, firstCode, te, templPath)
+	if err != nil {
+		t.Fatal(err)
+	}
 
-		thirdCode, err := renderService(svc, secondCode, te, templPath)
-		if err != nil {
-			t.Fatal(err)
-		}
+	if strings.Compare(firstCode, secondCode) != 0 {
+		t.Fatal("Generated code differs after regenerated with same definition\n" +
+			templPath + "\n" +
+			diff(firstCode, secondCode))
+	}
 
-		if strings.Compare(secondCode, thirdCode) != -1 {
-			t.Fatal("Generated code not longer after regenerated with additional service method\n" +
-				templPath + "\n" +
-				diff(secondCode, thirdCode))
-		}
+	svc.Methods = append(svc.Methods, allMethods[1])
 
-		// remove the first one rpc
-		svc.Methods = svc.Methods[1:]
+	thirdCode, err := renderService(svc, secondCode, te, templPath)
+	if err != nil {
+		t.Fatal(err)
+	}
 
-		forthCode, err := renderService(svc, thirdCode, te, templPath)
-		if err != nil {
-			t.Fatal(err)
-		}
+	if thirdCode <= secondCode {
+		t.Fatal("Generated code not longer after regenerated with additional service method\n" +
+			templPath + "\n" +
+			diff(secondCode, thirdCode))
+	}
 
-		if strings.Compare(thirdCode, forthCode) != 1 {
-			t.Fatal("Generated code not shorter after regenerated with fewer service method\n" +
-				templPath + "\n" +
-				diff(secondCode, thirdCode))
-		}
+	// remove the first one rpc
+	svc.Methods = svc.Methods[1:]
 
-		svc.Methods = allMethods
+	forthCode, err := renderService(svc, thirdCode, te, templPath)
+	if err != nil {
+		t.Fatal(err)
+	}
 
-		fifthCode, err := renderService(svc, forthCode, te, templPath)
-		if err != nil {
-			t.Fatal(err)
-		}
+	if forthCode >= thirdCode {
+		t.Fatal("Generated code not shorter after regenerated with fewer service method\n" +
+			templPath + "\n" +
+			diff(thirdCode, forthCode))
+	}
 
-		if strings.Compare(forthCode, fifthCode) != -1 {
-			t.Fatal("Generated code not longer after regenerated with additional service method\n" +
-				templPath + "\n" +
-				diff(secondCode, thirdCode))
-		}
+	svc.Methods = allMethods
+
+	fifthCode, err := renderService(svc, forthCode, te, templPath)
+	if err != nil {
+		t.Fatal(err)
 	}
-	testHandlerGeneration("NAME-service/handlers/server/server_handler.gotemplate")
-}
 
-func diff(a, b string) string {
-	return gentesthelper.DiffStrings(
-		a,
-		b,
-	)
+	if fifthCode <= forthCode {
+		t.Fatal("Generated code not longer after regenerated with additional service method\n" +
+			templPath + "\n" +
+			diff(forthCode, fifthCode))
+	}
 }
 
-func renderService(svc *svcdef.Service, prev string, te *gengokit.TemplateExecutor, templPath string) (string, error) {
-	h, err := handler.New(svc, strings.NewReader(prev), te.PackageName)
+func renderService(svc *svcdef.Service, prev string, te *gengokit.Executor, templPath string) (string, error) {
+	var prevFile io.Reader
```

Comment:
It represents the prevous version of a file, that is what this helper function is for.

In a pull #104 this function moves to the `handler` package and I can update the pull request to have a better comment for this helper function